HPXMLtoOpenStudio/resources/hvac_sizing.rb

```diff
@@ -1886,87 +1886,69 @@ def self.apply_hvac_ground_loop(hvac_sizing_values, weather, hvac_cooling)
-      if (bore_depth < min_bore_depth) && (num_bore_holes > 1)
-        num_bore_holes -= 1
-        bore_depth = (bore_length / num_bore_holes).floor
-      elsif bore_depth > 345
-        num_bore_holes += 1
-        bore_depth = (bore_length / num_bore_holes).floor
-      end
-    end
-
-    bore_depth = (bore_length / num_bore_holes).floor + 5
-
-    if num_bore_holes == 1
-      bore_config = 'single'
-    elsif num_bore_holes == 2
-      bore_config = 'line'
-    elsif num_bore_holes == 3
-      bore_config = 'line'
-    elsif num_bore_holes == 4
-      bore_config = 'rectangle'
-    elsif num_bore_holes == 5
-      bore_config = 'u-config'
-    elsif num_bore_holes > 5
-      bore_config = 'line'
-    end
-
-    # Test for valid GSHP bore field configurations
-    valid_configs = { 'single' => [1],
-                      'line' => [2, 3, 4, 5, 6, 7, 8, 9, 10],
-                      'l-config' => [3, 4, 5, 6],
-                      'rectangle' => [2, 4, 6, 8],
-                      'u-config' => [5, 7, 9],
-                      'l2-config' => [8],
-                      'open-rectangle' => [8] }
-    valid_num_bores = valid_configs[bore_config]
-    max_valid_configs = { 'line' => 10, 'l-config' => 6 }
-    unless valid_num_bores.include? num_bore_holes
-      # Any configuration with a max_valid_configs value can accept any number of bores up to the maximum
-      if max_valid_configs.keys.include? bore_config
-        max_num_bore_holes = max_valid_configs[bore_config]
-        num_bore_holes = max_num_bore_holes
-      else
-        # Search for first valid bore field
-        new_bore_config = nil
-        valid_configs.keys.each do |bore_config|
-          next unless valid_configs[bore_config].include? num_bore_holes
+    geothermal_loop = hvac_cooling.geothermal_loop
+    bore_spacing = geothermal_loop.bore_spacing
+    bore_diameter = geothermal_loop.bore_diameter
+    grout_conductivity = geothermal_loop.grout_conductivity
+    pipe_r_value = gshp_hx_pipe_rvalue(hvac_cooling)
+    nom_length_heat, nom_length_cool = gshp_hxbore_ft_per_ton(weather, hvac_cooling_ap, bore_spacing, bore_diameter, grout_conductivity, pipe_r_value)
 
-          new_bore_config = bore_config
-          break
-        end
-        if not new_bore_config.nil?
-          bore_config = new_bore_config
-        else
-          fail 'Could not construct a valid GSHP bore field configuration.'
+    loop_flow = geothermal_loop.loop_flow
+    if loop_flow.nil?
+      loop_flow = [1.0, UnitConversions.convert([hvac_sizing_values.Heat_Capacity, hvac_sizing_values.Cool_Capacity].max, 'Btu/hr', 'ton')].max.floor * 3.0
+    end
+
+    num_bore_holes = geothermal_loop.num_bore_holes
+    if num_bore_holes.nil?
+      num_bore_holes = [1, (UnitConversions.convert(hvac_sizing_values.Cool_Capacity, 'Btu/hr', 'ton') + 0.5).floor].max
+    end
+
+    min_bore_depth = UnitConversions.convert(24, 'm', 'ft')
+    max_bore_depth = UnitConversions.convert(384, 'm', 'ft')
```

Comment:
This line might be a good one to scrutinize. The 24 and 384 come directly from the limits in the g-function library JSON files. Would we ever see a borehole this deep in a residential application?

Reply:
Min we should keep, extrapolation would be a problem (especially on the short end, since ground temperatures fluctuate more at shallower depths). Max we discussed 500 ft (so 152.4 m). Based on needing a mining permit in NY. Sounds like theoretically we could go deeper (and maybe some homes do outside of NY), but this is a more reasonable assumption.